test/AssetPool.test.js

```diff
@@ -285,26 +328,57 @@ describe("AssetPool", () => {
 
     context("when called via approveAndCall", () => {
       const amount = to1e18(100)
+      const abiCoder = ethers.utils.defaultAbiCoder
 
-      beforeEach(async () => {
-        await collateralToken
-          .connect(underwriter1)
-          .approveAndCall(assetPool.address, amount, [])
+      context("when passed correct parameters", () => {
+        beforeEach(async () => {
+          const data = abiCoder.encode(["uint256"], [amount])
+
+          await collateralToken
+            .connect(underwriter1)
+            .approveAndCall(assetPool.address, amount, data)
+        })
+
+        it("should mint underwriter tokens to the caller", async () => {
+          expect(
+            await underwriterToken.balanceOf(underwriter1.address)
+          ).to.equal(amount)
+        })
+
+        it("should transfer deposited amount to the pool", async () => {
+          expect(await collateralToken.balanceOf(assetPool.address)).to.equal(
+            amount
+          )
+          expect(
+            await collateralToken.balanceOf(underwriter1.address)
+          ).to.be.equal(underwriterInitialCollateralBalance.sub(amount))
+        })
       })
 
-      it("should mint underwriter tokens to the caller", async () => {
-        expect(await underwriterToken.balanceOf(underwriter1.address)).to.equal(
-          amount
-        )
+      context("when passed incorrect parameters", () => {
```

Comment:
*when passed corrupted extradata parameter*? We are having only one parameter incorrect.

Reply:
Done as part of 485362d